src/targets/npm.ts

```diff
@@ -246,3 +263,95 @@ export class NpmTarget extends BaseTarget {
     this.logger.info('NPM release complete');
   }
 }
+
+/**
+ * Get the latest version for the given package.
+ */
+export async function getLatestVersion(
+  packageName: string,
+  npmConfig: NpmTargetOptions,
+  otp?: NpmPublishOptions['otp']
+): Promise<string | undefined> {
+  const args = ['info', packageName, 'version'];
+  const bin = NPM_BIN;
+
+  try {
+    const response = await withTempFile(filePath => {
+      // Pass OTP if configured
+      const spawnOptions: SpawnOptions = {};
+      spawnOptions.env = { ...process.env };
+      if (otp) {
+        spawnOptions.env.NPM_CONFIG_OTP = otp;
+      }
+      spawnOptions.env[NPM_TOKEN_ENV_VAR] = npmConfig.token;
+      // NOTE(byk): Use npm_config_userconfig instead of --userconfig for yarn compat
+      spawnOptions.env.npm_config_userconfig = filePath;
+      writeFileSync(
+        filePath,
+        `//registry.npmjs.org/:_authToken=\${${NPM_TOKEN_ENV_VAR}}`
+      );
+
+      // Disable output buffering because NPM/Yarn can ask us for one-time passwords
+      return spawnProcess(bin, args, spawnOptions, {});
+    });
+
+    if (!response) {
+      return undefined;
+    }
+
+    return response.toString().trim();
+  } catch {
+    return undefined;
+  }
+}
+/**
+ * Get the tag to use for publishing to npm.
+ */
+export async function getPublishTag(
```

Comment:
l: l: I'd either document that returning `undefined` later on causes npm to assign the latest tag automatically or we return `latest` here as a default. wdyt?